chains/ethereum/server/src/client.rs

```diff
@@ -142,28 +147,79 @@ where
             .as_u128())
     }
 
-    #[allow(clippy::unused_async)]
+    #[allow(clippy::unused_async, clippy::missing_errors_doc)]
     pub async fn coins(&self, _address: &Address, _block: &BlockIdentifier) -> Result<Vec<Coin>> {
         anyhow::bail!("not a utxo chain");
     }
 
-    pub async fn faucet(&self, address: &Address, param: u128) -> Result<Vec<u8>> {
-        // first account will be the coinbase account on a dev net
-        let coinbase = self.client.get_accounts().await?[0];
-        let address: H160 = address.address().parse()?;
-        let tx = TransactionRequest::new().to(address).value(param).from(coinbase);
-        Ok(self
-            .client
-            .send_transaction(tx, None)
-            .await?
-            .confirmations(2)
-            .await?
-            .context("failed to retrieve tx receipt")?
-            .transaction_hash
-            .0
-            .to_vec())
+    #[allow(clippy::single_match_else, clippy::missing_errors_doc)]
+    pub async fn faucet(
+        &self,
+        address: &Address,
+        param: u128,
+        private_key: Option<&str>,
+    ) -> Result<Vec<u8>> {
+        match private_key {
+            Some(private_key) => {
+                let chain_id = self.client.get_chainid().await?;
+                let address: H160 = address.address().parse()?;
+                let nonce = self
+                    .client
+                    .get_transaction_count(
+                        ethers::types::NameOrAddress::Address(H160::from_str(
+                            "0x3f1eae7d46d88f08fc2f8ed27fcb2ab183eb2d0e",
+                        )?),
+                        None,
+                    )
+                    .await?; //public key of faucet account
```

Comment:
This address should not be hard coded here, also the private-key can be an `[u8;32]` instead an string

Reply:
updated